### PR TITLE
Modify the script for dagw: add a fixed source:dagw, rename the origi…

### PR DESCRIPTION
**Pull request by TTTTao725**
…nal source: sub-source, move the sub-source to metadata, add domain filtering, add cleaning function.

Still working on the data sheets.

### CODE REVIEW

**Comment by TTTTao725** [comment]
languages: da

I'm not sure what else do we have :(

File: data-processing/scripts/convert_dagw_to_jsonlgz.py [hunk 1]
```diff
@@ -112,38 +114,65 @@ def reformat_dataset(ds: Dataset, num_proc: int) -> Dataset:
...
+    # Filter dataset based on domains
+    ds = filter_by_domains(ds, domain_list)
+
+    # Print out a sample
+    sample = ds[0]
+    print("Sample Record:", sample)
+
+    print("Start saving the dataset...")
+    # Save to jsonl.gz
+    ds.to_json("data.jsonl.gz", orient="records", lines=True, compression="gzip")
```
Comment (KennethEnevoldsen) [inline data-processing/scripts/convert_dagw_to_jsonlgz.py:167]: I believe this is only supported in the latest version of datasets so please update the dependency as well

File: docs/datasheets/augmented_dagw.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    license: "cc-by-4.0"
+  - name: nordjylland-news
+    description: "Dataset with text-summary pairs from Danish TV2 Nord news articles."
+    homepage: "https://huggingface.co/datasets/alexandrainst/nordjylland-news-summarization"
+    size: "75219 records"
+    license: "cc-by-4.0"
+```
+
+# Augmented DAGW
+This is a quick introduction and summarition of an augmented Danish Gigaword Dataset. Here we will describe the different components and how we did the simple cleaning.
```
Comment (KennethEnevoldsen) [inline docs/datasheets/augmented_dagw.md:43]:
```suggestion
This is a quick introduction and summarization of an augmented Danish Gigaword Dataset. Here we will describe the different components and how we did the simple cleaning.
```

**Review by KennethEnevoldsen — APPROVED** [review]
Datasheet looks very good. @peterbjorgensen let me know your thoughts on this

**Comment by peterbjorgensen** [comment]
> Datasheet looks very good. @peterbjorgensen let me know your thoughts on this

I notice that huggingface uses `---` to separate the yaml from the Markdown. I don't know if this a hard requirement for the huggingface hub and libraries to work correctly?

**Comment by peterbjorgensen** [comment]
So with respect to this decision https://github.com/centre-for-humanities-computing/danish-foundation-models/issues/266#issuecomment-2082092835 we need to update the dataset conversion script such that each sub-source is split into different datasets and each should have a separate dataset card.

File: docs/datasheets/augmented_dagw.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+- DAGW: https://github.com/centre-for-humanities-computing/danish-foundation-models/blob/main/data-processing/scripts/convert_dagw_to_jsonlgz.py
+- MeMo: https://github.com/centre-for-humanities-computing/danish-foundation-models/blob/main/data-processing/scripts/memo/convert_memo_to_jsonlgz.py
+- Scandi-Reddit: https://github.com/centre-for-humanities-computing/danish-foundation-models/blob/main/data-processing/scripts/convert_scandi_reddit_to_jsonlgz.py
+- Nordjylland-News: https://github.com/centre-for-humanities-computing/danish-foundation-models/blob/main/data-processing/scripts/convert_nordjyllandnews_to_jsonlgz.py
+## Data Cleaning
+
+1. First, we filtered out all the empty documents (based on `"text"`: either *empty* or totally *white space*) in DAGW: 162051 out of 673211 empty docs found in total.
+2. After combining all 4 components, we applied deduplication and URL block list to the augmented DAGW following this [filtering pipeline](https://github.com/centre-for-humanities-computing/danish-foundation-models/blob/main/data-processing/configs/2024-v1/README.md) with [Dolma](https://github.com/allenai/dolma).
```
Comment (peterbjorgensen) [inline docs/datasheets/augmented_dagw.md:96]: It does not hurt to delete empty documents, but I don't think we should do line level deduplication and URL blocking at this stage.

**Comment by KennethEnevoldsen** [comment]
> I notice that huggingface uses --- to separate the yaml from the Markdown. I don't know if this a hard requirement for the huggingface hub and libraries to work correctly?

I believe it is

**Comment by TTTTao725** [comment]
Thanks guys, I'll fix it this week :)

File: docs/datasheets/augmented_dagw.md (hunk 3 above)
Comment (TTTTao725) [reply docs/datasheets/augmented_dagw.md:96]: Ok, Peter, do you mean we should remove this second cleaning step at this stage?

Comment (peterbjorgensen) [reply docs/datasheets/augmented_dagw.md:96]: Yes, I don't think we should do the second cleaning step because it might mess up some of the content.

**Comment by TTTTao725** [comment]
Hi guys, how do you think of those markdowns I made automatically? 🤩

File: docs/datasheets/dagw-wikibooks.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+# Dataset Card for Wikibooks
+## Dataset Description
+- **Number of records:** 1319
+- **Languages:** Danish
+## Dataset Sturcture
+An example from the dataset looks as follows.
+```yaml
+{
+    'text': 'Spilinfo.
+Spillet er lavet af Blizzard Entertainme',
```
Comment (KennethEnevoldsen) [inline docs/datasheets/dagw-wikibooks.md:24]:
```suggestion
    'text': 'Spilinfo. Spillet er lavet af Blizzard Entertainme...
```
I would make is as so to make it clear that it is cut off

**Review by KennethEnevoldsen — APPROVED** [review]
Data sheets are looking better @TTTTao725 !

File: docs/datasheets/dagw-wikibooks.md (hunk 4 above)
Comment (TTTTao725) [reply docs/datasheets/dagw-wikibooks.md:24]: yes, will do. So what I have done before is simply printing out the first 50 characters, I'll filter out '\n' first and add '...' in the end

**Comment by KennethEnevoldsen** [comment]
@TTTTao725 let us get this merged in as well and then create a separate PR where we address some of the issues raised by the dataset validator #269.

**Comment by TTTTao725** [comment]
No problem!

**Pull request merged**